In [3]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
from sklearn.model_selection import train_test_split

# Loading Heart Dataset

In [2]:
df = pd.read_csv('heart_disease.csv')
df.sample(5)

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
162   41    1   1       120   157    0        1      182      0      0.0   
54    63    0   2       135   252    0        0      172      0      0.0   
159   56    1   1       130   221    0        0      163      0      0.0   
118   46    0   1       105   204    0        1      172      0      0.0   
82    60    0   2       102   318    0        1      160      0      0.0   

     slope  ca  thal  target  
162      2   0     2       1  
54       2   0     2       1  
159      2   0     3       1  
118      2   0     2       1  
82       2   1     2       1

# Feature descriptions
### Below is the group of features presents in the dataset segregated by their type (numerical, categorical, ordinal, binary)

# Binary
### sex (0 = female; 1 = male)
### fbs: Fasting blood sugar > 120 mg/dl
### exang: Exercise induced angina (0 = no; 1 = yes)
# Categorical
### cp: Chest pain type (0 = Asymptomatic angina; 1 = Atypical angina; 2 = Non-angina; 3 = Typical angina)
### restecg: Resting ECG (0 = Left ventricular hypertrophy; 1 = Normal; 2 = ST-T wave abnormality)
### slope: Slope of the peak exercise ST segment (0 = downsloping; 1 = upsloping; 2 = flat)
### thal: Thalium stress test result (0 = NA; 1 = Fixed defect; 2 = Normal; 3 = Reversible defect)
# Ordinal
### ca: number of major vessels (0-3) colored by flourosopy
# Numeric
### age
### oldpeak: ST depression induced by exercise relative to rest
### trestbps: Resting blood pressure
### chol: Serum cholestoral in mg/dl
### thalach: Maximum heart rate achieved during thalium stress test
# Target
### target: 1 = heart disease; 0 = no heart disease

In [3]:
bins = ['sex', 'fbs', 'exang']
cats = ['cp', 'restecg', 'slope', 'thal']
ords = ['ca']
nums = ['age', 'oldpeak', 'trestbps', 'chol', 'thalach']
target = ['target']

In [4]:
# one hot encoding the categorical variables for easy data exploration, you may reverse it later to numeric value for model training and testing

df.cp = df.cp.replace({0:'Asympt.', 1:'Atypical', 2:'Non', 3:'Typical'})
df.restecg = df.restecg.replace({0:'LV hyper', 1:'Normal', 2:'ST-T wave'})
df.slope = df.slope.replace({0:'down', 1:'up', 2:'flat'})
df.thal = df.thal.replace({0:'NA', 1:'Fixed', 2:'Normal', 3:'Revers.'})

In [5]:
df.describe()

age         sex    trestbps        chol         fbs     thalach  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean    54.366337    0.683168  131.623762  246.264026    0.148515  149.646865   
std      9.082101    0.466011   17.538143   51.830751    0.356198   22.905161   
min     29.000000    0.000000   94.000000  126.000000    0.000000   71.000000   
25%     47.500000    0.000000  120.000000  211.000000    0.000000  133.500000   
50%     55.000000    1.000000  130.000000  240.000000    0.000000  153.000000   
75%     61.000000    1.000000  140.000000  274.500000    0.000000  166.000000   
max     77.000000    1.000000  200.000000  564.000000    1.000000  202.000000   

            exang     oldpeak          ca      target  
count  303.000000  303.000000  303.000000  303.000000  
mean     0.326733    1.039604    0.729373    0.544554  
std      0.469794    1.161075    1.022606    0.498835  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000    0.000000  
50%      0.000000    0.800000    0.000000    1.000000  
75%      1.000000    1.600000    1.000000    1.000000  
max      1.000000    6.200000    4.000000    1.000000

# Load Iris Datset

# Feature descriptions
### The columns in this dataset are:

### Id
### SepalLengthCm
### SepalWidthCm
### PetalLengthCm
### PetalWidthCm
### Species

In [4]:
df_iris = pd.read_csv('iris_dataset.csv')

In [5]:
df_iris.describe()

sepal_length  sepal_width  petal_length  petal_width
count    105.000000   105.000000    105.000000   105.000000
mean       5.873333     3.050476      3.785714     1.204762
std        0.862941     0.454068      1.782793     0.778853
min        4.300000     2.000000      1.100000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.200000     1.300000
75%        6.400000     3.300000      5.100000     1.900000
max        7.900000     4.400000      6.900000     2.500000

In [6]:
df_iris.head()

sepal_length  sepal_width  petal_length  petal_width          species
0           5.1          2.5           3.0          1.1  Iris-versicolor
1           6.2          2.2           4.5          1.5  Iris-versicolor
2           5.1          3.8           1.5          0.3      Iris-setosa
3           6.8          3.2           5.9          2.3   Iris-virginica
4           5.7          2.8           4.1          1.3  Iris-versicolor

In [7]:
df_iris.columns

Index(['sepal_length', 'sepal_width', 'petal_length', 'petal_width',
       'species'],
      dtype='object')